features/0067-didcomms-diddoc-conventions/README.md

```diff
@@ -0,0 +1,180 @@
+# Did communication Did document conventions
+- Author: Tobias Looker (tobias.looker@mattr.global), Stephen Curran (swcurran@gmail.com)
+- Start Date: 04/15/2019
+
+## Status
+- Status: [![proposed](/proposed.png)](/README.md#hipe-lifecycle)
+- Status Date: 06/10/2019
+- Status Note: This revises the former [INDY HIPE](https://github.com/hyperledger/indy-hipe/pull/92)
+
+## Summary
+
+Explain the Did document conventions required to enable Did communications.
+
+## Motivation
+
+Standardization of these conventions is essential to promoting interoperability of Did communications.
+
+## Tutorial
+
+### Did documents
+
+A Did document is the associated data model to a Did, it contains important associated cryptographic information and a declaration of capabilities the Did supports. 
+
+Of particular interested to this RFC is the definition of [service endpoints](https://w3c-ccg.github.io/did-spec/#service-endpoints). The primary object of this RFC is to document the Did communication service type and describe the associated conventions.
+
+### Service Conventions
+
+As referenced above within the [Did specification](https://w3c-ccg.github.io/did-spec/#did-documents) lies a section called [service endpoints](https://w3c-ccg.github.io/did-spec/#service-endpoints), this section of the Did document is reserved for `any type of service the entity wishes to advertise, including decentralized identity management services for further discovery, authentication, authorization, or interaction`.
+
+When a Did document wishes to express support for Did communications, the following service definition is defined.
+
+```json
+{
+  "service": [{
+    "id": "did:example:123456789abcdefghi#did-communication",
+    "type": "did-communication",
+    "priority" : 0,
+    "recipientKeys" : [ "did:example:123456789abcdefghi#1" ],
+    "routingKeys" : [ "did:example:123456789abcdefghi#1" ],
+    "serviceEndpoint": "https://agent.example.com/"
+  }]
+}
+```
+
+- id : Required by the [Service Endpoints Spec](https://w3c-ccg.github.io/did-spec/#service-endpoints).
+- type : Required by the [Service Endpoints Spec](https://w3c-ccg.github.io/did-spec/#service-endpoints). It is mandatory that this field is set to `did-communication`
+- priority : This represents the priority of the service endpoint, used for distinction when multiple `did-communication` service endpoints are present in a single Did document. It is mandatory that this field is set to an unsigned integer with the default value of `0`.
+- recipientKeys : This is an array of [did key references](https://w3c-ccg.github.io/did-spec/#public-keys) used to denote the default recipients of an endpoint. (*note-1*)
+- routingKeys: This is an array of [did key references](https://w3c-ccg.github.io/did-spec/#public-keys), ordered from most destward to most srcward, used to denote the individual routing hops in between the sender and recipients. See [TBC]() for more information on how routing is intended to operate.
```

Comment:
@tplooker missing link for TBC

Reply:
@llorllale this is due to the routing RFC not being proposed in Aries yet, my intention was to correct this once the RFC is defined?